c++/src/kj/refcount.h

```diff
@@ -115,6 +132,93 @@ Own<T> Refcounted::addRefInternal(T* object) {
   return Own<T>(object, *refcounted);
 }
 
+template <typename T>
+Rc<T> Refcounted::addRcRefInternal(T* object) {
+  Refcounted* refcounted = object;
+  ++refcounted->refcount;
+  return Rc<T>(object);
+}
+
+template<typename T>
+class Rc: public Own<T> {
+  // Smart pointer for reference counted objects. 
+  // Can be used everywhere Own<T> is expected.
+  // NOTE: this will become primary way to access reference counted objects.
+
+public:
+  KJ_DISALLOW_COPY(Rc);
+  Rc() { }
+
+  template <typename... Params>
+  static Rc<T> create(Params&&...params) {
+    // Main factory function for Rc<T>. 
+    // NOTE: This will become global refcounted() function once migration is completed.
+    return Refcounted::addRcRefInternal(new T(kj::fwd<Params>(params)...));
+  }
+
+  inline Rc(Rc&& other) noexcept: Own<T>(kj::mv(other)) { }
+
+  template <typename U, typename = EnableIf<canConvert<U*, T*>()>>
+  inline Rc(Rc<U>&& other) noexcept : Own<T>(kj::mv(other)) { }
```

Comment:
For later, I wonder if we could eliminate (or at least obsolete) `kj::Own<kj::RefcountedWrapper<T>>` with a constructor here that accepts `kj::Own<T>&&` and wraps it internally as a `Refcounted`?

```cpp
kj::Own<Foo> foo;
kj::Rc<Foo> rcFoo = kj::mv(foo);
```

Since this would be eliminating `kj::addRef(*foo)` and replacing with the `addRef` on `kj::Rc`, the fact that `Foo` does not extend from `Refcounted` could be ok?